File: core/core/src/main/java/cn/netdiscovery/core/domain/response/HttpResponse.java
package cn.netdiscovery.core.domain.response;

import cn.netdiscovery.core.constants.ResponseCode;
import lombok.Data;

/**
 * 对外暴露接口返回的 HttpResponse
 * Created by tony on 2018/1/26.
 */
@Data
public class HttpResponse<T> {

    private int code;

    private String message;

    private T data;

    public static HttpResponse Ok = new HttpResponse(ResponseCode.Ok);
    public static HttpResponse Bad = new HttpResponse(ResponseCode.BadRequest);
    public static HttpResponse NotFound = new HttpResponse(ResponseCode.NotFound);
    public static HttpResponse InternalServerError = new HttpResponse(ResponseCode.InternalServerError);
    public static HttpResponse SpiderNotFound = new HttpResponse(ResponseCode.SpiderNotFound);

    public HttpResponse() {
    }

    public HttpResponse(ResponseCode code, String message) {
        this.code = code.getCode();
        this.message = message;
    }

    public HttpResponse(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public HttpResponse(T data) {
        this(ResponseCode.Ok);
        this.data = data;
    }

    public HttpResponse(ResponseCode code) {
        this.code = code.getCode();
        this.message = code.getMessage();
    }
}

File: core/core/src/main/kotlin/cn/netdiscovery/core/domain/response/HttpResponse.kt
package cn.netdiscovery.core.domain.response

import cn.netdiscovery.core.constants.ResponseCode

/**
 *
 * @FileName:
 *          cn.netdiscovery.core.domain.response.HttpResponse
 * @author: Tony Shen
 * @date: 2020-10-10 16:37
 * @version: V1.0 <描述当前版本功能>
 */
class HttpResponse<T> {

    private var code = 0
    private var message: String? = null
    private var data: T? = null

    constructor() {}

    constructor(code: ResponseCode, message: String?) {
        this.code = code.code
        this.message = message
    }

    constructor(code: Int, message: String?) {
        this.code = code
        this.message = message
    }

    constructor(data: T) : this(ResponseCode.Ok) {
        this.data = data
    }

    constructor(code: ResponseCode) {
        this.code = code.code
        message = code.message
    }

    companion object {
        var Ok: HttpResponse<*> = HttpResponse<Any?>(ResponseCode.Ok)
        var Bad: HttpResponse<*> = HttpResponse<Any?>(ResponseCode.BadRequest)
        var NotFound: HttpResponse<*> = HttpResponse<Any?>(ResponseCode.NotFound)
        var InternalServerError: HttpResponse<*> = HttpResponse<Any?>(ResponseCode.InternalServerError)
        var SpiderNotFound: HttpResponse<*> = HttpResponse<Any?>(ResponseCode.SpiderNotFound)
    }
}